src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,599 @@
+        """
+        MixFFN block (https://arxiv.org/pdf/2105.15203.pdf)
+        :param in_dim: input dimension
+        :param inter_dim: intermediate dimension
+        """
+
+        super().__init__()
+
+        self.fc1 = nn.Linear(in_dim, inter_dim)
+        self.dwconv = nn.Conv2d(in_channels=inter_dim, out_channels=inter_dim, kernel_size=3, stride=1, padding=1, groups=inter_dim)
+        self.fc2 = nn.Linear(inter_dim, in_dim)
+
+    def forward(self, x: torch.Tensor, h: int, w: int) -> torch.Tensor:
+        x = self.fc1(x)
+
+        b, _, c = x.shape
+        x = x.transpose(1, 2).view(b, c, h, w)
+        x = self.dwconv(x)
+        x = x.flatten(2).transpose(1, 2)
+
+        x = self.fc2(F.gelu(x))
+
+        return x
+
+
+class EncoderBlock(nn.Module):
+    def __init__(self, dim: int, head: int, sr_ratio: int, dpr: float):
+        """
+        A single encoder block (https://arxiv.org/pdf/2105.15203.pdf)
+        :param dim: embedding dimension
+        :param head: number of attention heads
+        :param sr_ratio: the reduction ratio of the efficient self-attention
+        :param dpr: drop-path ratio
+        """
+
+        super().__init__()
+
+        self.attn = EfficientSelfAttention(dim, head, sr_ratio)
+
+        self.drop_path = DropPath(dpr) if dpr > 0.0 else nn.Identity()
+
+        self.norm1 = nn.LayerNorm(dim)
+        self.norm2 = nn.LayerNorm(dim)
+
+        self.mlp = MixFFN(in_dim=dim, inter_dim=dim * 4)
+
+    def forward(self, x: torch.Tensor, h: int, w: int) -> torch.Tensor:
+        x = x + self.drop_path(self.attn(self.norm1(x), h, w))
+        x = x + self.drop_path(self.mlp(self.norm2(x), h, w))
+
+        return x
+
+
+class MiTBackBone(nn.Module):
+    def __init__(
+        self,
+        embed_dims: list,
+        encoder_layers: list,
+        eff_self_att_reduction_ratio: list,
+        eff_self_att_heads: list,
+        overlap_patch_size: list,
+        overlap_patch_stride: list,
+        overlap_patch_pad: list,
+        in_channels: int,
+    ):
+        """
+        Mixed Transformer backbone encoder (https://arxiv.org/pdf/2105.15203.pdf)
+        :param embed_dims: the patch embedding dimensions (number of output channels in each encoder stage)
+        :param encoder_layers: the number of encoder layers in each encoder stage
+        :param eff_self_att_reduction_ratio: the reduction ratios of the efficient self-attention in each stage
+        :param eff_self_att_heads: number of efficient self-attention heads in each stage
+        :param overlap_patch_size:  the patch size of the overlapping patch embedding in each stage
+        :param overlap_patch_stride:  the patch stride of the overlapping patch embedding in each stage
+        :param overlap_patch_pad:  the patch padding of the overlapping patch embedding in each stage
+        :param in_channels:  number of input channels
+        """
+
+        super().__init__()
+
+        assert (
```

Comment:
Rule of thumb: assert in unit tests, if/raise in the code. 
WHY: https://github.com/Deci-AI/super-gradients/pull/599#discussion_r1116999292

Reply:
Changed to if/raise